op-batcher/batcher/sync_actions.go

```diff
@@ -43,6 +43,11 @@ func (s syncActions) TerminalString() string {
 		"SyncActions{blocksToPrune: %d, channelsToPrune: %d, clearState: %v, blocksToLoad: %v}", s.blocksToPrune, s.channelsToPrune, cs, btl)
 }
 
+func isZero[T eth.L2BlockRef | eth.L1BlockRef](x T) bool {
+	var y T
+	return (x == y)
+}
```

Comment:
For a long time I wanted to add `IsZero() bool` methods to the block refs types and `BlockID` 😂 maybe that's nicer than this package local solution.

And note that your helper works with any type really (if you prefer to keep it at all)
```suggestion
func isZero[T any](x T) bool {
	var y T
	return (x == y)
}
```

Reply:
Actually it doesn't work for literally `any` type, because that would include incomparable types like functions. So we do actually need to narrow it here.

Reply:
But does it compile? As long as a type is initializable with `var x T` and you don't use `isZero` on silly types like functions, it should just be fine. If you ever passed it a type for which the expression doesn't make sense, it should give a compilation error.

Reply:
It does not compile.

Reply:
maybe?
```suggestion
func isZero[T comparable](x T) bool {
	var y T
	return (x == y)
}
```

Reply:
yep that works [65e7726](https://github.com/ethereum-optimism/optimism/pull/16298/commits/65e7726a340c694737ce48d78d6706b5ceec9889)